tests/test_google_connect.py

```diff
@@ -0,0 +1,321 @@
+from app.database.models import OAuthCredentials
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from googleapiclient.http import HttpMock
+from app.routers.user import create_user
+
+
+@pytest.fixture
+def google_events_mock():
+    return [
+        {
+            "kind": "calendar#event",
+            "etag": "somecode",
+            "id": "somecode",
+            "status": "confirmed",
+            "htmlLink": "https://www.google.com/calendar/event?eid=somecode",
+            "created": "2021-01-13T09:10:02.000Z",
+            "updated": "2021-01-13T09:10:02.388Z",
+            "summary": "some title",
+            "creator": {
+                "email": "someemail",
+                "self": True
+            },
+            "organizer": {
+                "email": "someemail",
+                "self": True
+            },
+            "start": {
+                "dateTime": "2021-02-25T13:00:00+02:00"
+            },
+            "end": {
+                "dateTime": "2021-02-25T14:00:00+02:00"
+            },
+            "iCalUID": "somecode",
+            "sequence": 0,
+            "reminders": {
+                "useDefault": True
+            }
+        },
+        {
+            "kind": "calendar#event",
+            "etag": "somecode",
+            "id": "somecode",
+            "status": "confirmed",
+            "htmlLink": "https://www.google.com/calendar/event?eid=somecode",
+            "created": "2021-01-13T09:10:02.000Z",
+            "updated": "2021-01-13T09:10:02.388Z",
+            "summary": "some title to all day event",
+            "creator": {
+                "email": "someemail",
+                "self": True
+            },
+            "organizer": {
+                "email": "someemail",
+                "self": True
+            },
+            "start": {
+                "date": "2021-02-25"
+            },
+            "end": {
+                "date": "2021-02-25"
+            },
+            "iCalUID": "somecode",
+            "sequence": 0,
+            "location": 'somelocation',
+            "reminders": {
+                "useDefault": True
+            }
+        }
+    ]
+
+
+@pytest.fixture
+def mock_cred():
+    cred = Credentials(
+        token="somecode",
+        refresh_token="somecode",
+        token_uri="some_uri",
+        client_id="somecode",
+        client_secret="some_secret",
+        expiry=datetime(2021, 1, 28)
+    )
+
+    cred.refresh = safe_refresh
+
+    return cred
+
+
+def safe_refresh(*args):
+    print('refreshed')
+
+
+def safe_run_local_server(*args, **kwargs):
+    print('running server')
+
+
+@pytest.mark.usefixtures("user")
+@pytest.mark.usefixtures("session")
+@pytest.mark.usefixtures("google_events_mock")
+def test_push_events_to_db(google_events_mock, user, session):
+    assert google_connect.push_events_to_db(google_events_mock, user, session)
+
+
+@pytest.mark.usefixtures("user")
```

Comment:
You can separate it with comma, or not mention it at all :)

Reply:
About which line you're talking about?